source/localizable/tutorial/service.md

```diff
@@ -0,0 +1,289 @@
+For Super Rentals, we want to add the ability to display maps of where each rental is located.  To implement this feature, we will take advantage of several Ember concepts:
+
+1. A component to display a map on each rental listing.
+2. A service to keep a cache of rendered maps to use in different places in the application.
+3. A utility function to create a map from Google's map API
+
+We'll work on this feature outside-in, meaning we'll start with the display of the map and work our way back to the actual access of the Google Maps API.
+
+
+### Display Maps With a Component
+
+For our maps component, all we'll require is a simple component interface that we can simply drop into a template.
+The component will render a map pinned to whatever location we provide it.
+Lets start by appending the component as we want it to the end of the `rental-listing` template, and provide it with the city of the current rental.
+
+```app/templates/components/rental-listing.hbs
+<h2>{{rental.title}}</h2>
+<p>Owner: {{rental.owner}}</p>
+<p>Type: {{rental-property-type rental.type}} - {{rental.type}}</p>
+<p>Location: {{rental.city}}</p>
+<p>Number of bedrooms: {{rental.bedrooms}}</p>
+{{#if isImageShowing}}
+  <p>
+    <img alt={{rental.type}} src={{rental.image}} width="500"/>
+  </p>
+  <button {{action "imageHide"}}>Hide image</button>
+{{else}}
+  <button {{action "imageShow"}}>Show image</button>
+{{/if}}
+{{location-map location=rental.city}}
+```
+
+Next step is to get our page compiling by using Ember CLI to generate this component.
+
+```shell
+ember g component location-map
+```
+
+The Ember CLI component generator will create for us a test file, a component JavaScript file, and a template.
+To think about what we want our component to do, will write a testcase.
+In this case we want our Google Maps service to provide us with the map display, so all we want our component to do is take the output of our maps service, which is a map element, and append it to an element in the component's template.
+
+All we want to do in our component is make sure its appending the output of the maps service to an inner map container element.
+To limit the test to validating just this behavior we'll take advantage of the registration api to provide a mock amps service.
+In the mock service, define a method that will fetch the map based on location, called `getMapElement`.
+
+```tests/integration/components/location-map-test.js
+import { moduleForComponent, test } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+import Ember from 'ember';
+
+let MockMapsService = Ember.Service.extend({
+  getMapElement(location) {
+    this.set('calledWithLocation', location);
+    return document.createElement('div');
```

Comment:
because the maps service returns an element that gets appended to the dom in the component.  I can add some text around it.